instrumentation/opentelemetry-instrumentation-asgi/tests/test_asgi_custom_headers.py

```diff
@@ -130,6 +130,10 @@ async def test_http_custom_request_headers_in_span_attributes(self):
                 (b"Regex-Test-Header-1", b"Regex Test Value 1"),
                 (b"regex-test-header-2", b"RegexTestValue2,RegexTestValue3"),
                 (b"My-Secret-Header", b"My Secret Value"),
+                (
+                    b"non-utf8-header",
+                    b"Moto Z\xb2",
+                ),
```

Comment:
Great! It would be nice to add more test cases here, for example:
1- When the header key is a non utf-8 string
2- When both header key and header value are non utf-8 strings.